zappa/handler.py

```diff
@@ -480,6 +484,126 @@ def handler(self, event, context):
                 logger.error("Cannot find a function to process the triggered event.")
             return result
 
+        # This is an HTTP-protocol API Gateway event or Lambda url event with payload format version 2.0
+        elif event.get("version", None) and event["version"] == "2.0":
+            try:
+                time_start = datetime.datetime.now()
+
+                script_name = ""
+                host = event.get("headers", {}).get("host")
+                if host:
+                    if "amazonaws.com" in host:
+                        logger.debug("amazonaws found in host")
+                        # The path provided in th event doesn't include the
+                        # stage, so we must tell Flask to include the API
+                        # stage in the url it calculates. See https://github.com/Miserlou/Zappa/issues/1014
+                        script_name = "/" + settings.API_STAGE
+                else:
+                    # This is a test request sent from the AWS console
+                    if settings.DOMAIN:
+                        # Assume the requests received will be on the specified
+                        # domain. No special handling is required
+                        pass
+                    else:
+                        # Assume the requests received will be to the
+                        # amazonaws.com endpoint, so tell Flask to include the
+                        # API stage
+                        script_name = "/" + settings.API_STAGE
+
+                base_path = getattr(settings, "BASE_PATH", None)
+                environ = create_wsgi_request_from_v2_event(
+                    event,
+                    script_name=script_name,
+                    base_path=base_path,
+                    trailing_slash=self.trailing_slash,
+                    binary_support=settings.BINARY_SUPPORT,
+                    context_header_mappings=settings.CONTEXT_HEADER_MAPPINGS,
+                )
+
+                # We are always on https on Lambda, so tell our wsgi app that.
+                environ["HTTPS"] = "on"
+                environ["wsgi.url_scheme"] = "https"
+                environ["lambda.context"] = context
+                environ["lambda.event"] = event
+
+                # Execute the application
+                with Response.from_app(self.wsgi_app, environ) as response:
+                    response_body = None
+                    response_is_base_64_encoded = False
+                    if response.data:
+                        if (
```

Comment:
There's a BINARY_SUPPORT change I'm trying to get in, where this will be changed.

https://github.com/zappa/Zappa/pull/1155

Reply:
@monkut thanks for notifying this. how do we usually solve this kind of issue? do zappa have rc branch? or should I rebase and move my PR base branch to your feature branch?

Reply:
Can you rebase off of the BINARY_SUPPORT branch?  I'm working to get the BINARY_SUPPORT feature branch in on the next release (Targeted at mid November).

Reply:
@monkut sure, I've done it. however I'll let my current PR base pointing to master. feel free to change if needed (just in case to create stacked PR)